Supersig: like multisig, but with superpowers! 💥

decentration opened the pull request:
This is Decentration's first grant proposal to Web3. We view this simple, suitable and potentially pervasively used pallet as a great opportunity to develop an ongoing relationship with Web3 Foundation, given that we have shared and aligned interests.

**key features that improve on the mulitsig pallet**

- add/remove members, 
- supersig account is fixed, not a composite of its accounts, like multsig. 
- enumerated quorum (simpleMajority, superMajority), instead of fixed x/n threshold. 
- option to create Super Beneficiary, this beneficary only needs 2/n threshold, while any transaction without Super Beneficary must have superMajority. 

These palletsimprovements have already created a lot of interest and feedback from people in the community, so we believe it's a valuable coordination of resources. 

We also believe in building things people need rather over-engineering without validation. The simplicity of this pallet is a great example of this. 

### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying?

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $50,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for > $100k Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [x] I have read and understood the [FAQs](https://github.com/w3f/Grants-Program/blob/master/docs/faq.md), [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md) and [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/announcement-guidelines.md).
- [x] A BTC or Ethereum (DAI/USDT) address for the payment of the milestones is provided inside the application.
- [x] I have read and acknowledge the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).
- [ ] I prefer the discussion of this application to be in a private Element/Matrix channel. My username is: ``@_______``

### How Did You Hear About our grants program?

- [ ] Social Media
- [ ] Hackathon
- [x] Personal Recommendation
- [x] Substrate Builders Program
- [x] Investor/VC
- [ ] Online Search
- [x] Other: Substrate Seminar Crowdcast

laboon reviewed (approved):
1. Excellent use-case. This is something that I personally would find useful (maybe that is not the best definition of "excellent", since I am a bit biased towards my own use cases of course, but that's at least one person that will use it)
2. Good technical description
3. Team is capable of developing a pallet

Approved

Noc2 reviewed (changes requested):
Thanks for the application. Sounds interesting. Something to be aware of: We usually sign at the beginning only one grant per team and we just approved: https://github.com/w3f/Grants-Program/pull/949 So could you explain what your relationship with them is? Keep in mind that we can also make an exception to this rule. Also, you still need to sign the CLA and could you add the delivery tables to milestone 1 and 2, including the default deliverables of the template 0a-0d.

decentration commented:
> only one grant per team and we just approved: #949 So could you explain what your relationship with them is? Keep in mind that we can also make an exception to this rule. 

Hi @Noc2, indeed. Rusty Crewmates are an engineer DAO. They have core educational objectives, which have been funded in #949, but they also train and provide engineering talent to other chains, teams and communities for Substrate. Edgeware and Kabocha being their first partners. Nathan (mentioned in this proposal) is a Rusty Crewmate, who went through their training. He is one of the full time FTE's in our milestone 1. He is not mentioned in #949.  The only crossover in the two applications is Timothee who is busy on #949, but providing very light support in this application. I thought i put his name here because he deserves some credit, but not making up the FTE calculation. 

In the future Rusty Crewmates will be peppered around many Substrate chain teams, and they set a high bar, so beneficial for all!

Noc2 reviewed (changes requested):
Thanks for the update. This sounds good to me and I’m generally willing to go ahead with it. However, milestone 3 is currently very loosely defined. Could you add more details to it, for example [benchmarking](https://docs.substrate.io/v3/runtime/benchmarking/#:~:text=Instead%2C%20Substrate%20expects%20benchmarking%20to,and%20fees%20can%20be%20returned) or what are the necessary updates? Otherwise I would suggest simply removing milestone 3 and immediately deliver everything with milestone 2.

Noc2 commented:
And there seems to be an issue with the formatting of your last milestone.

decentration commented:
Hi David @Noc2, thanks for feedback. I updated milestone 3 to add more colour. It's an essential part of our process, because it allows us to get feedback from both milestone 1 (pallet) and milestone 2 (UI), and to make detailed and/or holistic changes. (Vague because of unknowns). I also added Benchmarking task to milestone 1. Let me know if anything is not detailed enough?

Noc2 reviewed (approved):
Sounds good to me. I’m happy to go ahead with it and will mark the application as ready for review.

semuelle commented on applications/pallet_supersig.md:
Can you please remove the template text from all `Article` deliverables and replace them with what you are actually planning to publish?

semuelle reviewed (approved):
LGTM

tdelabro commented:
Hey,
I'm a bit late to the party, but I want to thank you all for entrusting Kabocha, Ramsey, and RustyCrewmates with this grant.
We are smoothing some rough edges on the RPC side right now.
I think you are going to like the result!

ramseykes commented:
I can confirm I am @decentration, and this is my main account.

keeganquigley commented:
Hey @decentration any updates on milestone 2? How are you coming along?

keeganquigley commented:
Thanks for the update @decentration yep sounds good, just let us know if you think the deliveries will be delayed by more than 2 weeks.

keeganquigley commented:
Hi @decentration just checking back in with a friendly reminder. If M2 is complete I would recommend submitting it first and then submit M3 in a separate PR when it's done. Otherwise if you are still working on it, please submit an [amendment](https://github.com/w3f/Grants-Program#2-application-review) to have the committee approve an extension. Thanks!

decentration commented:
Hi Keegan, 

Just to clarify, do you mean I can submit to the committee an ammended for time? Becuase we don’t need to amend the application for any changes of what will be delivered. I would prefer not to request any amendment if I don’t have to. 

Tbh I wasn’t aware of the time urgency, if you look at M1, we completed the work a long time before the request for payment of the milestone. If we knew prior that there was an urgency to deliver within a 3 month period we would have done so. This grant application was approved much longer than 3 months ago.

Apologies if there is an expectation of delivery.  This work is most definitely active and a priority!

Could you tell me what the situation with the Builders Programme? The substrate stackexchange is very help, but more support in relation to Polkadot js apps might speed this up. 

Kind regards,

Ramsey

keeganquigley commented:
Thanks for the update @decentration. That's not a problem, I was mostly just asking for a progress update and whether or not separating them was a possibility. 

Yes, to answer your question, the amendment would just be for updating the timeline, which generally is easy to get the committee to approve.

But no worries, we mostly just want to make sure the project is active and being worked on. If you think there will be significant delays, please let us know and consider submitting an amendment. Thanks!
